### PR TITLE
[clang] Enable Wenum-constexpr-conversion also in system headers and …

### PR DESCRIPTION
**Pull request by carlosgalvezp**
…macros

As per review comments on https://reviews.llvm.org/D150226, we should allow for one more release before turning this warning into a hard error, by making it visible in system headers and macros, so that people are aware of it and can work on it.

### CODE REVIEW

File: clang/include/clang/Basic/DiagnosticASTKinds.td
Comment (Fznamznon) [inline clang/include/clang/Basic/DiagnosticASTKinds.td:408]: Do we need a test for this change?

Comment (carlosgalvezp) [reply clang/include/clang/Basic/DiagnosticASTKinds.td:408]: Yes, will fix!

Comment (carlosgalvezp) [reply clang/include/clang/Basic/DiagnosticASTKinds.td:408]: Done.

**Comment by carlosgalvezp** [comment]
I would need some help with the failing pre-merge test, I don't really understand why that failure would be related to this patch...

**Comment by carlosgalvezp** [comment]
It seems checks are broken on trunk, I see commits merged with failing pre-merge tests. They seem to be unrelated to this patch though.

Is there anything else you'd like fixed before merging? @dwblaikie @AaronBallman @shafik

**Comment by dwblaikie** [comment]
> It seems checks are broken on trunk, I see commits merged with failing pre-merge tests. They seem to be unrelated to this patch though.
> 
> Is there anything else you'd like fixed before merging? @dwblaikie @AaronBallman @shafik

Nothing further from me, at least. But maybe wait for approval from one of the other two.

**Comment by carlosgalvezp** [comment]
Friendly ping @AaronBallman @shafik

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Comment by smeenai** [comment]
https://github.com/boostorg/mpl/issues/69 is still a problem, unfortunately.

**Comment by shafik** [comment]
Thank you for doing this work.

**Pull request merged**
